FirebaseSessions/Sources/SessionStartEvent.swift

```diff
@@ -256,44 +256,87 @@ class SessionStartEvent: NSObject, GDTCOREventDataObject {
     -> firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype {
     var subtype: firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype
 
-    #if os(iOS) && !targetEnvironment(macCatalyst)
-      switch mobileSubtype {
-      case CTRadioAccessTechnologyGPRS:
-        subtype = firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype_GPRS
-      case CTRadioAccessTechnologyEdge:
-        subtype = firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype_EDGE
-      case CTRadioAccessTechnologyWCDMA:
-        subtype = firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype_CDMA
-      case CTRadioAccessTechnologyCDMA1x:
-        subtype = firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype_CDMA
-      case CTRadioAccessTechnologyHSDPA:
-        subtype = firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype_HSDPA
-      case CTRadioAccessTechnologyHSUPA:
-        subtype = firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype_HSUPA
-      case CTRadioAccessTechnologyCDMAEVDORev0:
-        subtype = firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype_EVDO_0
-      case CTRadioAccessTechnologyCDMAEVDORevA:
-        subtype = firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype_EVDO_A
-      case CTRadioAccessTechnologyCDMAEVDORevB:
-        subtype = firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype_EVDO_B
-      case CTRadioAccessTechnologyeHRPD:
-        subtype = firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype_EHRPD
-      case CTRadioAccessTechnologyLTE:
-        subtype = firebase_appquality_sessions_NetworkConnectionInfo_MobileSubtype_LTE
-      default:
+    // swift(>=5.9) implies Xcode 15+
+    #if swift(>=5.9)
```

Comment:
Could we add a comment here about why this is duplicated? So we can use the (os(xrOS) macro)

Reply:
Done!

Reply:
Done!